core/plans/microvm_create_update.go

```diff
@@ -76,13 +81,15 @@ func (p *microvmCreateOrUpdatePlan) Create(ctx context.Context) ([]planner.Proce
 	}
 
 	// MicroVM provider create
-	if err := p.addStep(ctx, microvm.NewCreateStep(p.vm, ports.Provider)); err != nil {
+	if err := p.addStep(ctx, microvm.NewCreateStep(p.vm, provider)); err != nil {
 		return nil, fmt.Errorf("adding microvm create step: %w", err)
 	}
 
 	// MicroVM provider start
-	if err := p.addStep(ctx, microvm.NewStartStep(p.vm, ports.Provider, microVMBootTime)); err != nil {
-		return nil, fmt.Errorf("adding microvm start step: %w", err)
+	if provider.Capabilities().Has(models.StartCapability) {
+		if err := p.addStep(ctx, microvm.NewStartStep(p.vm, provider, microVMBootTime)); err != nil {
```

Comment:
With the above suggestion it would looks like this:
```suggestion
	if !provider.Capabilities().Has(models.AutostartCapability) {
		if err := p.addStep(ctx, microvm.NewStartStep(p.vm, provider, microVMBootTime)); err != nil {
```
which I think is more accurate as we want to start if the vm doesn't autostart on creation.

(still feel free to ignore and just mark as resolved)

Reply:
Changed :)